### ISSUE TITLE
emit value conflicts with hidden fields but not definitions

### ISSUE THREAD

**Issue by cueckoo** [cue-lang/cue#583]
_Originally opened by @jba in https://github.com/cuelang/cue/issues/583_

### What version of CUE are you using (`cue version`)?

<pre>
$ cue version
cue version v0.2.2 linux/amd64 
</pre>

### Does this issue reproduce with the latest release?
Yes.

### What did you do?

The spec describes hidden fields and definitions as similar ("Definitions and hidden fields are not emitted when converting a CUE program to data and are never required to be concrete."). But `cue eval -c` on this file:
```
_x: {a: 1, b: 2}
[for k, v in _x {alpha: k, beta: v}]
```
produces a "conflicting values" error, while this file succeeds:
```
#x: {a: 1, b: 2}
[for k, v in #x {alpha: k, beta: v}]
```

**Comment by cueckoo** [comment]
_Original reply by @myitcv in https://github.com/cuelang/cue/issues/583#issuecomment-723534274_

This appears to be working at `tip` FWIW.

**Comment by cueckoo** [comment]
_Original reply by @mpvl in https://github.com/cuelang/cue/issues/583#issuecomment-728850738_

This works in v0.3, which is a completely different implementation. It doesn't make sense to fix this in v0.2 at this point.
